libnmstate/state.py

```diff
@@ -745,10 +745,25 @@ def _sort_ip_addresses(self):
                 )
 
     def _capitalize_mac(self):
+        """
+        The MAC address is upper case but case insensitive. This is
+        capitalizing the MAC address. In addition, this is unifying the output
+        format MAC for SR-IOV VFs.
+        """
         for ifstate in self.interfaces.values():
             mac = ifstate.get(Interface.MAC)
             if mac:
                 ifstate[Interface.MAC] = mac.upper()
+            if ifstate[Interface.TYPE] == InterfaceType.ETHERNET:
+                vfs = (
+                    ifstate.get(Ethernet.CONFIG_SUBTREE, {})
+                    .get(Ethernet.SRIOV_SUBTREE, {})
+                    .get(Ethernet.SRIOV.VFS_SUBTREE, [])
+                )
+                for vf in vfs:
+                    vf_mac = vf.get(Ethernet.SRIOV.VFS.MAC_ADDRESS)
+                    if mac:
+                        vf[Ethernet.SRIOV.VFS.MAC_ADDRESS] = vf_mac.upper()
```

Comment:
Please move these lines to a function in `appliers/sriov.py`, no need to make the bloated `state.py` bigger.

Reply:
Done, thanks!